samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -874,14 +937,30 @@ static void light_lightness_range_set_unack(struct bt_mesh_model *model,
 					    struct bt_mesh_msg_ctx *ctx,
 					    struct net_buf_simple *buf)
 {
+	u16_t min, max;
 	struct net_buf_simple *msg = model->pub->msg;
 	struct light_lightness_state *state = model->user_data;
 
-	state->lightness_range_min = net_buf_simple_pull_le16(buf);
-	state->lightness_range_max = net_buf_simple_pull_le16(buf);
+	min = net_buf_simple_pull_le16(buf);
+	max = net_buf_simple_pull_le16(buf);
 
 	/* Here, Model specification is silent about tid implementation */
 
+	if (min <= max && min != 0 && max != 0) {
+		state->status_code = 0x00;
+
+		state->lightness_range_min = min;
+		state->lightness_range_max = max;
+	} else if (max < min) {
+		/* The provided value for Range Max cannot be set */
+		state->status_code = 0x02;
+	}
+
+	if (min == 0) {
+		/* The provided value for Range Min cannot be set */
+		state->status_code = 0x01;
```

Comment:
consider using a simple define for this, e.g 
#define STATE_STATUS_SUCCESS 0x00
#define STATE_STATUS_FAILED 0x01

Reply:
There are 3 values as of now, as per 7.2..... so I will define

#define STATE_STATUS_SUCCESS 0x00
#define CANNOT_SET _RANGE_MIN 0x01
#define CANNOT_SET _RANGE_MAX 0x02